parsec/core/invite_claim.py

```diff
@@ -64,8 +64,14 @@ class InviteClaimInvalidTokenError(InviteClaimError):
     pass
 
 
-def generate_invitation_token():
-    return token_hex(8)
+def generate_invitation_token(length=6):
+    """Generate a token of 6 digits, whith the first one different than 0.
+
+    Equivalent of `randint(100000, 999999)`.
+    """
+    mini = int("1".ljust(length, "0"))
```

Comment:
pourquoi pas `10 ** (length - 1)` et `(10 ** length) - 1`?

et pourquoi mini == 100000 ? de ce que je comprend ça veut dire par exemple qu'on n'aura jamais le code `012345` par exmple...
j'imagine que c'est pour ne pas avoir à faire de post-traitement sur le nombre généré aléatoirement (qui peut faire moins de 6 digits autrement), mais sans aucun commentaire de justification je trouve que ça ressemble à petite une vulnérabilité ;-)

Reply:
> pourquoi pas 10 ** (length - 1) et (10 ** length) - 1?

Ca marche aussi, mais la garanti de taille fixe est un peu moins évidente. J'aurais peut etre juste du mettre:
```python
mini = 100000
maxi = 999999
```
et enlever la paramétrisation (qui ne sert sans doute à rien). T'en penses quoi?

> et pourquoi mini == 100000 ? 
> mais sans aucun commentaire de justification je trouve que ça ressemble à petite une vulnérabilité ;-)

Y'a quand meme la docstring qui le dit explicitement juste au dessus :sweat_smile: 

    Generate a token of 6 digits, whith the first one different than 0.
   
Et oui la raison c'est pour éviter les confusions du genre `"012345" != "12345"`. Si tu trouves que ca sert a rien je peux enlever la contrainte et formater le résultat, e.g `format(12345, "06d")`.